### PR TITLE
add additional supportedplatform enum values for future / expansion use

### PR DESCRIPTION
**Pull request by davidkline-ms**
## Overview
Add iOS, Web and Lumin to the SupportedPlatform enums. This will allow future platform extensions to be created for these without needing to recompile the core assemblies.

This is related to the work by @Alexees (#4358) that has been moved to being under consideration for the next major MRTK release.

Also added Mac and Linux editor platforms to address #4601.

image

### CODE REVIEW

File: Assets/MixedRealityToolkit/Definitions/Utilities/SupportedPlatforms.cs [hunk 1]
```diff
@@ -16,6 +16,9 @@ public enum SupportedPlatforms
         LinuxStandalone = 1 << 2,
         WindowsUniversal = 1 << 3,
         WindowsEditor = 1 << 4,
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Definitions/Utilities/SupportedPlatforms.cs]: Do we want to add `MacEditor` and `LinuxEditor` at this time as well?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Definitions/Utilities/SupportedPlatforms.cs]: we might indeed

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Definitions/Utilities/SupportedPlatforms.cs]: added

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityInputSystemProfile.asset [hunk 2]
```diff
@@ -66,7 +66,7 @@ MonoBehaviour:
       reference: Microsoft.MixedReality.Toolkit.Input.InputSimulationService, Microsoft.MixedReality.Toolkit.Services.InputSimulation.Editor
     componentName: Input Simulation Service
     priority: 0
-    runtimePlatform: 16
+    runtimePlatform: 208
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityInputSystemProfile.asset]: Is this change intentional?

Comment (julenka) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityInputSystemProfile.asset]: Ah, yes it is: it's binary OR of Linux, Mac, Windows editors. (1 << 4) | (1 << 6) | (1 << 7). Thank you!
